### PR TITLE
Add max player limit to rooms

### PR DESCRIPTION
**Pull request by izakfr**
Closes: #98

### CODE REVIEW

File: server/pkg/models/room/errors.go [hunk 1]
```diff
@@ -7,7 +7,11 @@ import (
 )
 
 var errRoomNotFound = errortypes.New(errortypes.NotFound, "Room error: not found")
+
+var errRoomFull = errortypes.New(errortypes.BadRequest, "Room error: room has maximum number of players.")
```
Comment (jbsummoner) [inline server/pkg/models/room/errors.go:11]: I think we should have some synergy between variable name and error message

File: server/pkg/models/room/entity.go [hunk 2]
```diff
@@ -9,10 +9,13 @@ import (
...
 type Room struct {
-	ID      uuid.UUID   `json:"id"`
-	Name    string      `json:"name"`
-	Players []uuid.UUID `json:"players"`
-	GameID  *uuid.UUID  `json:"game_id"`
+	ID   uuid.UUID `json:"id"`
+	Name string    `json:"name"`
+
+	Players        []uuid.UUID `json:"players"`
+	PlayerCountMax int         `json:"player_count_max"`
```
Comment (jbsummoner) [inline server/pkg/models/room/entity.go:16]: WDYT about 
`PlayerLimit int `json:"player_limit"``

**Review by jbsummoner — APPROVED** [review]
looks good small nits :) 🦾

File: server/pkg/models/room/actions.go [hunk 3]
```diff
@@ -22,10 +22,11 @@ func (r *RequestNewRoom) PerformAction() (*Room, error) {
...
 
 	room := &Room{
-		ID:      uuid.New(),
-		Name:    r.Name,
-		Players: make([]uuid.UUID, 0),
-		mux:     &sync.RWMutex{},
+		ID:             uuid.New(),
+		Name:           r.Name,
+		Players:        make([]uuid.UUID, 0),
+		PlayerCountMax: 2,
```
Comment (jbsummoner) [inline server/pkg/models/room/actions.go:28]: could we hoist this to the top of the file and give it like const maybe

File: server/pkg/models/room/errors.go (hunk 1 above)
Comment (jbsummoner) [reply server/pkg/models/room/errors.go:11]: would it beneficial to return the limit as well?

**Comment by jbsummoner** [comment]
This looks good to go :) 

Screenshot 2022-12-16 at 11 16 01 PM

File: server/pkg/models/room/errors.go (hunk 1 above)
Comment (izakfr) [reply server/pkg/models/room/errors.go:11]: yes, good point, will update

File: server/pkg/models/room/entity.go (hunk 2 above)
Comment (izakfr) [reply server/pkg/models/room/entity.go:16]: that's a much better name

**Pull request merged**